Sources/iOSClientExposurePlayback/Services/ProgramService.swift

```diff
@@ -265,10 +261,20 @@ extension ProgramService {
     
     
     fileprivate func validate(program: Program, callback: @escaping (ExposureContext.Warning.ProgramService?, String?) -> Void) {
-        self.provider.validate(entitlementFor: program.assetId, environment: self.environment, sessionToken: self.sessionToken) { [weak self] validation, error in
-            
-            print(" Provider validae ")
-            
+        
+        // Get next program start time & pass to the entitlement check 
+        let programStartTimeInDateFormat = program.startTime?.toDate()
+        var timeToPass = program.startTime
+        
+        if let addingSecondToPRogramStartTime = programStartTimeInDateFormat?.addingTimeInterval(TimeInterval( 1 * 01 )) {
+            let formatter = DateFormatter()
+            formatter.locale = Locale(identifier: "en_GB")
+            formatter.timeZone = TimeZone(abbreviation: "UTC")
+            formatter.dateFormat = "yyyy-MM-dd'T'HH:mm:ss.SSS'Z'"
+            timeToPass =  formatter.string(from: addingSecondToPRogramStartTime)
+        }
+        
+        self.provider.validate(entitlementFor: program.assetId, environment: self.environment, sessionToken: self.sessionToken, programStartTime: timeToPass ) { [weak self] validation, error in
```

Comment:
As mentioned in the other PR, I think we shouldn't pass the date argument if program.startTime is in the past, or we should pass the current time instead. In my implementation I'm passing the highest of program.startTime + 1ms and the current time. The exposure server is supposed to default to the latter, but I found that not to work for one of the CU/BU SVT streams I tested with (although I think it was the streams fault).

Reply:
I don't quite understand that argument , when will be the scenario that  next program `program.startTime` will be in the past & sdk is doing an entitlement validation ?

Reply:
> I don't quite understand that argument , when will be the scenario that next program `program.startTime` will be in the past & sdk is doing an entitlement validation ?

When you're not at the live point. For example a user could do something like pausing the stream for a long time, then play.

Reply:
Maybe that scenario can't be possible for the ios app. It is for the js-player though.

Reply:
True , you have a point. I think it's better to pass the `program.startTime + 1ms and the current time`

Reply:
Yeah, in the js-player I did `max(currentTime, program + startTime + 1ms)`